src/main/groovy/com/cloudcard/photoDownloader/TouchNetClient.groovy

```diff
@@ -58,6 +63,72 @@ class TouchNetClient {
         log.info("        TouchNet Origin ID : $originId")
     }
 
+    @Override
+    String getSystemName() {
+        return "TouchNet"
+    }
+
+    @Override
+    void putPhoto(String accountId, String photoBase64) {
+        ensureSession()
+
+        Map request = [
+                SessionID: sessionId,
+                AccountID: accountId,
+                PhotoBase64: photoBase64,
+                ForcePrintedFlag: false
+        ]
+
+        TouchNetResponse response = doApiRequest("Account Photo Approve", "account/photo/approve", request)
+        if (!response.success) {
+            String message = response.getErrorMessage() ?: "Unknown TouchNet error"
+
+            if (isRateLimitError(message)) {
+                for (int i = 1; i <= MAX_TRIES; i++) {
+                    RETRY_DELAY_MS * i
+                    log.warn("Rate limit hit, retrying in ${RETRY_DELAY_MS}ms (attempt $i/$MAX_TRIES)")
+                    Thread.sleep(RETRY_DELAY_MS)
```

Comment:
In more detail than my above comment, we should implement the wait time as a function that takes in the number of tries and performs the wait - something like:

```
void exponentialWait(int tryCount) {
    int waitTime = RETRY_DELAY_MS * 2^tryCount // todo could add random jitter for +/- half of RETRY_DELAY_MS
    log.debug("waiting $waitTime before retry")
    Thread.sleep(waitTime)
}
```